Improve Haskell version for shorter edit distance between `then().then().then().世()`

igrep opened the pull request.

igrep committed:
Improve Haskell version for shorter edit distance between `then().then().then().世()`

MikuroXina commented on RADWIMPS.hs:
```diff
@@ -1,14 +1,9 @@
+  where
+    then' :: () -> IO ()
+    then' () = putStr "前"
+    世 () = putStrLn "世"
```
```suggestion
    世 :: () -> IO ()
    世 () = putStrLn "世"
```

MikuroXina commented on RADWIMPS.hs:
```diff
@@ -1,14 +1,9 @@
+    then' :: () -> IO ()
+    then' () = putStr "前"
+    世 () = putStrLn "世"
+    (.) = (>>)
```
```suggestion
    (.) :: IO a -> IO b -> IO b
    (.) = (>>)
```

igrep committed:
Specify type signatures

fumieval commented on RADWIMPS.hs:
```diff
@@ -1,14 +1,13 @@
-世 :: IO String
-世 = return "世"
+then' :: () -> IO ()
+then' () = putStr "前"
```
thenをCPS変換してみては？`(.)`を上書きする必要がなくなります

igrep replied:
なるほどさすが。やってみました + ついでに、他の言語のバージョンと同様 `radwimps`というキーワードをちゃんと含めることにしました。
https://github.com/approvers/RADWIMPS/pull/32/commits/298af8f5c5c0c2154fec6cecd40d785666bc81f3

igrep committed:
Specify the radwimps function as the other versions do

And to avoid redefining `(.)`. Thanks: https://github.com/approvers/RADWIMPS/pull/32#discussion_r448185761
